content/automate/ManageIQ/Transformation/Infrastructure/VM/Common.class/__methods__/checkpoweredon.rb

```diff
@@ -0,0 +1,41 @@
+module ManageIQ
+  module Automate
+    module Transformation
+      module Infrastructure
+        module VM
+          module Common
+            class CheckPoweredOn
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                if @handle.root['service_template_transformation_plan_task'].blank?
+                  task = @handle.vmdb(:service_template_transformation_plan_task).find_by(:id => @handle.root['service_template_transformation_plan_task_id'])
+                  vm = task.source if task.present?
+                else
+                  task = @handle.root['service_template_transformation_plan_task']
+                  vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id)) if task.present?
+                end
```

Comment:
We should break this down into methods which would make the logic easier to read.

Move this first if-else block into a method that returns both `task` and `vm`:
```ruby
def task_and_vm
  if @handle.root['service_template_transformation_plan_task'].blank?
    task = @handle.vmdb(:service_template_transformation_plan_task).find_by(:id => @handle.root['service_template_transformation_plan_task_id'])
    vm = task.source if task.present?
  else
    task = @handle.root['service_template_transformation_plan_task']
    vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id)) if task.present?
  end

  return task, vm
end
```

Then the first part of main becomes:
```ruby
task, vm = task_and_vm
```

If you think this is too much change it could be in a followup refactoring, but we need to move away for having large single method scripts in general.

Reply:
The code is identical in `checkpoweredon.rb` and `poweron.rb`. I'd propose moving `task_and_vm` to a embedded utility method. It's a bigger change that may require another PR. I'd go for a followup refactoring.